databridge.examples/databridge.examples.aas-jsonata-http/src/test/java/org/eclipse/digitaltwin/basyx/databridge/examples/aasjsonatahttp/test/TestAASUpdater.java

```diff
@@ -0,0 +1,197 @@
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.eclipse.basyx.aas.registration.memory.InMemoryRegistry;
+import org.eclipse.basyx.components.aas.AASServerComponent;
+import org.eclipse.basyx.components.aas.configuration.AASServerBackend;
+import org.eclipse.basyx.components.aas.configuration.BaSyxAASServerConfiguration;
+import org.eclipse.basyx.components.configuration.BaSyxContextConfiguration;
+import org.eclipse.digitaltwin.basyx.databridge.aas.configuration.factory.AASPollingConsumerDefaultConfigurationFactory;
+import org.eclipse.digitaltwin.basyx.databridge.core.component.DataBridgeComponent;
+import org.eclipse.digitaltwin.basyx.databridge.core.configuration.factory.RoutesConfigurationFactory;
+import org.eclipse.digitaltwin.basyx.databridge.core.configuration.route.core.RoutesConfiguration;
+import org.eclipse.digitaltwin.basyx.databridge.httppolling.configuration.factory.HttpProducerDefaultConfigurationFactory;
+import org.eclipse.digitaltwin.basyx.databridge.jsonata.configuration.factory.JsonataDefaultConfigurationFactory;
+import org.eclipse.digitaltwin.basyx.databridge.timer.configuration.factory.TimerDefaultConfigurationFactory;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.HttpResponse;
+import org.mockserver.model.HttpStatusCode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A test of aas-jsonata-http
+ * @author rana
+ *
+ */
+public class TestAASUpdater {
+
+	private static Logger logger = LoggerFactory.getLogger(TestAASUpdater.class);
+	private static AASServerComponent aasServer;
+	private static BaSyxContextConfiguration aasContextConfig;
+	private static InMemoryRegistry registry = new InMemoryRegistry();
+	private static DataBridgeComponent updater;
+	private static int PORT = 8080;
+	private static ClientAndServer clientServer;
+	
+	@BeforeClass
+	public static void setUp() throws Exception {
+
+		configureAndStartAasServer();
+		
+		mockServerConfig();
+	
+		configureAndStartUpdaterComponent();
+	}
+	
+	@AfterClass
+	public static void tearDown() throws Exception {
+		updater.stopComponent();
+		
+		aasServer.stopComponent();
+	}
+	
+	@Test
+	public void CheckPropertyValueA() throws  InterruptedException, ClientProtocolException, IOException {
+		
+		String expected_value = "103.5585973";
+		
+		startHTTPServer(getResponseBody(expected_value));
```

Comment:
This here sets the value to the mocked server, and while retrieving returns the same value, as explained in the above comment.

Reply:
changed